src/concrete/ml/quantization/quantized_module.py

```diff
@@ -337,26 +337,31 @@ def forward(
         q_x = to_tuple(self.quantize_input(*x))
 
         if debug and fhe == "disable":
-            debug_value_tracker: Optional[
-                Dict[str, Dict[Union[int, str], Optional[ONNXOpInputOutputType]]]
+            debug_value_tracker: Dict[
+                str, Dict[Union[int, str], Optional[ONNXOpInputOutputType]]
             ] = {}
             for (_, layer) in self.quant_layers_dict.values():
                 layer.debug_value_tracker = debug_value_tracker
-            result = self.quantized_forward(*q_x, fhe="disable")
+            q_y_pred = self.quantized_forward(*q_x, fhe="disable")
             for (_, layer) in self.quant_layers_dict.values():
                 layer.debug_value_tracker = None
-            return result, debug_value_tracker
+            # De-quantize the output predicted values
+            y_pred = self.dequantize_output(
```

Comment:
are you sure we want to de-quant here ? I believe the debug option was made to get the quantized values no the floating points ? cc @andrei-stoian-zama

Reply:
I'm not the one that uses this feature the most honestly so not sure !

Reply:
Maybe @andrei-stoian-zama can tell us

Reply:
so actually this makes me think that it's better to return floatings point for debug_values, or then we move this debug code to `quantized_forward` !

Reply:
> debug_values are integers yes. I see your point about changing forward to return floats, but the forward was returning integers before.. maybe for some good reason, like client/server ?

Pretty sure we don't use the debug mode. Also this change didn't break any test so idk.

Reply:
actually, we do have a complete test for debugging interediary values : `test_brevitas_intermediary_values` .. looking at it quickly, I'm surprised it didn't break anything, it seems to be expecting some quantized values at some point, so worth to take a quick look at it I would say 😶